### ISSUE TITLE
Support MkDocs `1.4.0` `hooks:` config option in `mkdocs.yml`

### ISSUE THREAD

**Issue by HonkingGoose** [squidfunk/mkdocs-material#4420]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

I know that MkDocs `1.4.0` is really new, so I'm not expecting you to support this new feature right away. 😉

MkDocs `1.4.0` [^mkdocs] has a new `hooks:` [^hooks] config option. You can use the `hooks:` option in the `mkdocs.yml` file. This new feature is not compatible with Material for MkDocs right now.

When I run `mkdocs serve` it raises this error:

```
pipenv run mkdocs serve
INFO     -  Building documentation...
WARNING  -  Config value: 'hooks'. Warning: Unrecognised configuration name: hooks
Aborted with 1 Configuration Warnings in 'strict' mode!
make: *** [Makefile:27: serve] Error 1
```

From the error message it looks like `hooks:` is not supported yet by Material for MkDocs.

[^mkdocs]: https://github.com/mkdocs/mkdocs/releases/tag/1.4.0
[^hooks]: https://www.mkdocs.org/user-guide/configuration/#hooks

### Use Cases

Getting support for `hooks:` in Material for MkDocs _may_ allow us to drop the `mkdocs-simple-hooks` dependency.

**Comment by squidfunk** [comment]
Thanks for reporting. Does the warning also appear with another theme than Material for MkDocs? Are you sure that you upgraded to 1.4.0? I'm not sure what the theme would need to support here, as the warnings comes from MkDocs itself.

**Comment by oprypin** [comment]
Yes this doesn't make sense to me. How did you conclude from this error message that it's something about mkdocs-material?
How about `pipenv run mkdocs --version` - I'd expect it to show that you didn't actually upgrade MkDocs

**Comment by squidfunk** [comment]
I've tried reproducing it, to no avail. It's likely that the upgrade to 1.4.0 didn't work. Please make sure MkDocs 1.4.0 is installed. If the error persists and you can clearly pin it to Material for MkDocs, please reopen.

**Comment by HonkingGoose** [comment]
Thank you @oprypin and @squidfunk for looking at my "bug report". ❤️ 

I messed up the upgrade to the MkDocs `1.4.0` package. That's why `hooks:` was "broken" for me. 🙈 I'm sorry about spending your time on my bad bug report! 😞 

I managed to get the new `hooks:` feature working on the Renovate docs repository now. 😉
